### PR TITLE
CS-174: Feature/check proof purpose

### PR DESCRIPTION
**Pull request by PManaras**

### CODE REVIEW

**Comment by nitin-vavdiya** [comment]
We need to add test cases inDidDocumentTest to build did document with authentication and assertionMethod. We also need to assert once we have a new did document (Verify the new did document).

**Review by nitin-vavdiya — CHANGES_REQUESTED** [review]

**Comment by PManaras** [comment]
> We need to add test cases inDidDocumentTest to build did document with authentication and assertionMethod. We also need to assert once we have a new did document (Verify the new did document).

there is actually a test, but it's in MIW :-D

File: src/main/java/org/eclipse/tractusx/ssi/lib/model/did/DidDocumentBuilder.java [hunk 1]
```diff
@@ -66,19 +78,75 @@ public DidDocumentBuilder verificationMethod(VerificationMethod verificationMeth
...
             DidDocument.CONTEXT,
             DidDocument.DEFAULT_CONTEXT,
             DidDocument.ID,
             id.toString(),
             DidDocument.VERIFICATION_METHOD,
-            verificationMethods));
+            verificationMethods);
+
+    HashMap<String, Object> entries = new HashMap<>(requiredEntries);
+    if (assertionMethod != null && !assertionMethod.isEmpty())
```
[the remark this answers is not in the document]
Comment (PManaras) [reply src/main/java/org/eclipse/tractusx/ssi/lib/model/did/DidDocumentBuilder.java]: refactored

File: src/main/java/org/eclipse/tractusx/ssi/lib/model/did/DidDocumentBuilder.java [hunk 2]
```diff
@@ -66,19 +78,75 @@ public DidDocumentBuilder verificationMethod(VerificationMethod verificationMeth
...
+      entries.put("assertionMethod", assertionMethod);
```
[the remark this answers is not in the document]
Comment (PManaras) [reply src/main/java/org/eclipse/tractusx/ssi/lib/model/did/DidDocumentBuilder.java]: refactored

**Review by nitin-vavdiya — APPROVED** [review]

**Pull request merged**